addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -84,13 +87,72 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, fmt.Errorf("failed to enable mirroring the storagecluster %q in namespace %q in managed cluster. Error %v", scr.Name, scr.Namespace, err)
 	}
 
+	err = r.createS3(ctx)
+	if err != nil {
+		klog.Error(err, "Failed to create ODR S3 resources")
+		return ctrl.Result{}, err
+	}
+
 	err = r.createVolumeReplicationClass(ctx, &mirrorPeer)
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("failed to create volumereplicationclass: %v", err)
 	}
 	return ctrl.Result{}, nil
 }
 
+func (r *MirrorPeerReconciler) createS3(ctx context.Context) error {
+	var err error
+	bucket := "odrbucket"
+	namespace := "openshift-dr-system"
+	if newNamespace := os.Getenv("ODR_NAMESPACE"); newNamespace != "" {
+		namespace = newNamespace
+	}
+
+	noobaaOBCNamespace := &corev1.Namespace{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "namespace",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		},
+	}
+	err = r.SpokeClient.Get(ctx, types.NamespacedName{Name: namespace}, noobaaOBCNamespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.Info("Could not find ODR OBC namespace")
+			return err
+		} else {
+			klog.Error(err, "Failed to get OBC namespace")
+			return err
+		}
+	}
+
+	noobaaOBC := &obv1alpha1.ObjectBucketClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      bucket,
+			Namespace: namespace,
+		},
+		Spec: obv1alpha1.ObjectBucketClaimSpec{
+			StorageClassName: "openshift-storage.noobaa.io",
+		},
+	}
+	err = r.SpokeClient.Get(ctx, types.NamespacedName{Name: bucket, Namespace: namespace}, noobaaOBC)
```

Comment:
I dont see any RBAC for ObjectBucketClaim resource

Reply:
Do we need one? I was able to resolve the permission issue by applying the two yaml changes below:

- https://github.com/red-hat-storage/odf-multicluster-orchestrator/pull/59/commits/ebed15d18cb772d602d0dcc56f222dfd8f8e4f21#diff-082c74b796220f9065744a958d99d64e363f0a21b299808ab8eefacd1bfcc031R14
- https://github.com/red-hat-storage/odf-multicluster-orchestrator/pull/59/commits/ebed15d18cb772d602d0dcc56f222dfd8f8e4f21#diff-3e13b7441890fb8cb2e1934a40a475f3c94e555309068e2d5d7f478b2e4119acR22

Not sure if this is the best way to do it, though.